tesseract/tesseract_visualization/src/visualization_loader.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * @file visualization_loader.h
+ * @brief Visualization Loader
+ *
+ * @author Levi Armstrong
+ * @date July 29, 2020
+ * @version TODO
+ * @bug No known bugs
+ *
+ * @copyright Copyright (c) 2020, Southwest Research Institute
+ *
+ * @par License
+ * Software License Agreement (Apache License)
+ * @par
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * @par
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <tesseract_common/macros.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_PUSH
+#include <console_bridge/console.h>
+TESSERACT_COMMON_IGNORE_WARNINGS_POP
+
+#include <tesseract_visualization/visualization_loader.h>
+
+const std::string TESSERACT_IGNITION_LIBRARY = "/snap/tesseract-ignition/current/opt/ros/melodic/lib/"
+                                               "libTesseractIgnitionVisualization.so";
+const std::string TESSERACT_IGNITION_CLASS = "tesseract_ignition::TesseractIgnitionVisualization";
+
+namespace tesseract_visualization
+{
+VisualizationLoader::VisualizationLoader()
+  : library_path_(TESSERACT_IGNITION_LIBRARY), derived_class_(TESSERACT_IGNITION_CLASS)
+{
+  createLoader(TESSERACT_IGNITION_LIBRARY);
+}
+
+VisualizationLoader::VisualizationLoader(const std::string& library_path, const std::string& derived_class)
+  : library_path_(library_path), derived_class_(derived_class)
+{
+  createLoader(library_path);
+}
+
+Visualization::Ptr VisualizationLoader::get()
+{
+  // Dynamically load visualization library
+  if (loader_)
+  {
+    try
+    {
+      if (loader_->isClassAvailable<tesseract_visualization::Visualization>(derived_class_))
+      {
+#ifndef CLASS_LOADER_LESS_4_0_0
```

Comment:
Is this checking if class_loader is less than 4.0.0? I think you intended to check if it is at least 4.0.0. I am not too familiar with this syntax so I may just be misreading it too.

Reply:
In the cmake file it checks if the version is less than 0.4.0 and if so it set a the compiler definition `CLASS_LOADER_LESS_0_4_0`. Then here I check if it is not defined, then if not defined then the version of class loader is at least version 0.4.0 or greater. This is because kinetic is still using boost shared pointer instead of std::shared_ptr so this will not work on kinetic.

Reply:
Ah, that makes sense. All seems good to me then